### ISSUE TITLE
MD051: Link fragments should be valid [Context: "[Stream.\_\_init__](#2221-__init__)"]

### ISSUE THREAD

**Issue by wuwentao** [DavidAnson/markdownlint#984]
markdown title content as below:

```markdown
##### 2.2.2.1 \_\_init__
```

TOC link can works with  `#2221-__init__`

but reference text with this URL report error:

```
[Stream.\_\_init__](#2221-__init__)
```

current version:

```
markdownlint-cli2 v0.10.0 (markdownlint v0.31.1)
```

error log:
```
MD051/link-fragments Link fragments should be valid [Context: "[Stream.\_\_init__](#2221-__init__)"]
```

**Comment by DavidAnson** [comment]
Thanks, I suspect the escaping slashes are getting included incorrectly.

https://dlaa.me/markdownlint/#%25m%23%20Issue%20984%0A%0A%23%23%202.2.2.1%20%5C_%5C_init__%0A%0A%5BLink%5D(%232221-__init__)%0A
